### PR TITLE
fix AttributeException

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#11462

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 1]
```diff
@@ -175,13 +175,13 @@ class HttpTransport(
 
     @abc.abstractmethod
     def send(self, request, **kwargs):
-        # type: (PipelineRequest, Any) -> PipelineResponse
+        # type: (PipelineRequest, Any) -> HTTPResponseType
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:178]: Input type is also HTTPRequestType

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 2]
```diff
@@ -175,13 +175,13 @@ class HttpTransport(
...
         """Send the request using this HTTP sender.
 
         :param request: The pipeline request object
         :type request: ~azure.core.pipeline.PipelineRequest
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:182]: azure.core.pipeline.transport.HttpRequest

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 3]
```diff
@@ -175,13 +175,13 @@ class HttpTransport(
...
         :return: The pipeline response object.
-        :rtype: ~azure.core.pipeline.PipelineResponse
+        :rtype: ~azure.core.pipeline.transport.HttpResponse
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:184]: Are AsyncHttpTransport typing/docstring correct?

File: sdk/core/azure-core/tests/test_stream_generator.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    HttpTransport,
+)
+from azure.core.pipeline import Pipeline, PipelineResponse
+from azure.core.pipeline.transport._requests_basic import StreamDownloadGenerator
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+def test_connetion_error_response():
```
Comment (lmazuel) [inline sdk/core/azure-core/tests/test_stream_generator.py:18]: typo

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py (hunk 2 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:182]: The input is correct. It is ~azure.core.pipeline.PipelineRequest

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py (hunk 3 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:184]: It does not have docstring. I added.

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:184]: We don't have typing/docstring for AsyncHttpTransport. So it is not incorrect. :)

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:184]: Might be the opportunity to add them, but yeah optional :p

**Pull request merged**